src/main/java/uk/gov/hmcts/reform/em/stitching/batch/RemoveOldDocumentTaskTasklet.java

```diff
@@ -36,18 +36,14 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
         Instant currentDate = Instant.now();
         Instant pastDate = currentDate.minus(numberOfDays, ChronoUnit.DAYS);
 
-        LOGGER.debug("Remove the DocumentTask before the {}", pastDate);
+        LOGGER.info("Remove the DocumentTask before the {}", pastDate);
 
-        List<Long> documentTaskIds = documentTaskRepository.findAllByCreatedDate(pastDate);
-
-        LOGGER.info("Number of DocumentTask rows retrieved {}", documentTaskIds.size());
+        List<Long> documentTaskIds = documentTaskRepository.findAllByCreatedDate(pastDate, numberOfRecords);
 
         if (CollectionUtils.isNotEmpty(documentTaskIds)) {
-            documentTaskIds.stream()
-                    .limit(numberOfRecords)
-                    .forEach(documentTaskId -> documentTaskRepository.deleteById(documentTaskId));
+            LOGGER.info("Number of DocumentTask rows retrieved {}", documentTaskIds.size());
+            documentTaskRepository.deleteAllById(documentTaskIds);
```

Comment:
should be fine but better to see executed SQLs

Reply:
Yes. I can confirm - When we delete any DocumentTask row. Bundle, CallBack are also deleted as they have CascadeType.ALL